### PR TITLE
Update Radio content

### PR DESCRIPTION
**Pull request by alex-ju**
### :pushpin: Summary

This PR updates the Radio page content following the model Checkbox as a reference.

In a separate commit, I've added a few fixes to the Checkbox content and have consistently ordered sections: group, field, base.

🥺 [Have a look at me, the Radio page](https://hds-website-git-alex-ju-radios-docs-hashicorp.vercel.app/components/form/radio) 

### :link: External links

Jira ticket: [HDS-1288](https://hashicorp.atlassian.net/browse/HDS-1288)

***

### 👀 Reviewer's checklist:

- [ ] +1 Percy if applicable
- [ ] Confirm that PR has a changelog update via [Changesets](https://github.com/changesets/changesets) if needed

:speech_balloon: Please consider using [conventional comments](https://conventionalcomments.org/) when reviewing this PR.

### CODE REVIEW

**Review by Dhaulagiri — APPROVED** [review]
👏 👏 👏

File: website/docs/components/form/checkbox/partials/code/component-api.md [hunk 1]
```diff
@@ -2,9 +2,54 @@
...
+    Appends a ’Required’ indicator next to the legend text and sets the `required` attribute on the controls when user input is required.
+  </C.Property>
+  <C.Property @name="isOptional" @type="boolean">
+    Appends an ’Optional’ indicator next to the legend text when user input is optional.
+  </C.Property>
+</Doc::ComponentApi>
+
+#### Contextual components
+
+Legend, group of fields and error content are passed to the group as yielded components, using the `Legend`, `Checkbox::Field`, `Error` keys.
```
Comment (jorytindall) [inline website/docs/components/form/checkbox/partials/code/component-api.md:28]:
```suggestion
Legend, groups of fields, and error content are passed to the group as yielded components, using the `Legend`, `Checkbox::Field`, and `Error` keys.
```

File: website/docs/components/form/checkbox/partials/code/component-api.md [hunk 2]
```diff
@@ -2,9 +2,54 @@
...
+
+The group of elements is automatically wrapped in a `<fieldset>` element.
+
+<Doc::ComponentApi as |C|>
+  <C.Property @name="<[G].Legend>" @type="yielded component">
+    It is an (optional) container that yields its content inside the `<legend>` element. The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Legend`](/components/form/base-elements/) component.
```
Comment (jorytindall) [inline website/docs/components/form/checkbox/partials/code/component-api.md:34]:
```suggestion
    An optional container that yields its content inside the `<legend>` element. The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Legend`](/components/form/base-elements/) component.
```

File: website/docs/components/form/checkbox/partials/code/component-api.md [hunk 3]
```diff
@@ -2,9 +2,54 @@
...
+  </C.Property>
+  <C.Property @name="<[G].HelperText>" @type="yielded component">
+    It is a container that yields its content inside the "helper text" block (at group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::HelperText`](/components/form/base-elements/) component.
```
Comment (jorytindall) [inline website/docs/components/form/checkbox/partials/code/component-api.md:37]:
```suggestion
    A container that yields its content inside the "helper text" block (at the group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::HelperText`](/components/form/base-elements/) component.
```

File: website/docs/components/form/checkbox/partials/code/component-api.md [hunk 4]
```diff
@@ -2,9 +2,54 @@
...
+    <br/><br/>
+    The `id` attribute of the element is automatically generated.
+  </C.Property>
+  <C.Property @name="<[G].Checkbox::Field>" @type="yielded component">
+    It is used to yield one or more fields inside the group. For details about its API check the `Checkbox::Field` component above.
```
Comment (jorytindall) [inline website/docs/components/form/checkbox/partials/code/component-api.md:42]:
```suggestion
    Used to yield one or more fields inside the group. For details about its API check the `Checkbox::Field` component above.
```

File: website/docs/components/form/checkbox/partials/code/component-api.md [hunk 5]
```diff
@@ -2,9 +2,54 @@
...
+  </C.Property>
+  <C.Property @name="<[G].Error>" @type="yielded component">
+    It is a container that yields its content inside the "error" block (at group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Error`](/components/form/base-elements/) component.
```
Comment (jorytindall) [inline website/docs/components/form/checkbox/partials/code/component-api.md:45]:
```suggestion
    A container that yields its content inside the "error" block (at group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Error`](/components/form/base-elements/) component.
```

File: website/docs/components/form/checkbox/partials/code/how-to-use.md [hunk 6]
```diff
@@ -342,6 +193,130 @@ There may be use cases in which you need to create a checkbox "group" that conta
...
+```handlebars
+<Hds::Form::Checkbox::Field as |F|>
+  <F.Label>Enable cost estimation</F.Label>
+  <F.HelperText>With this option enabled you will receive an approximate cost estimation.</F.HelperText>
+</Hds::Form::Checkbox::Field>
+```
+
+#### Extra content in label and helper text
+
+The `Label` and `HelperText` contextual components used in the "field" are yielding their content. This means you can also pass structured content.
```
Comment (jorytindall) [inline website/docs/components/form/checkbox/partials/code/how-to-use.md:248]:
```suggestion
The `Label` and `HelperText` contextual components used in the "field" yield their content, meaning you can also pass structured content.
```

File: website/docs/components/form/radio/partials/code/component-api.md [hunk 7]
```diff
@@ -1,95 +1,113 @@
...
+  <C.Property @name="isOptional" @type="boolean">
+    Appends an `Optional` indicator next to the legend text when user input is optional.
   </C.Property>
 </Doc::ComponentApi>
 
-##### Contextual components
+#### Contextual components
 
-Label, helper text and error content are passed to the field as yielded components, using the `Label`, `HelperText`, `Error` keys.
+Legend, group of fields and error content are passed to the group as yielded components, using the `Legend`, `Radio::Field`, `Error` keys.
```
Comment (jorytindall) [inline website/docs/components/form/radio/partials/code/component-api.md:30]:
```suggestion
Legend, group of fields, and error content are passed to the group as yielded components, using the `Legend`, `Radio::Field`, and `Error` keys.
```

File: website/docs/components/form/radio/partials/code/component-api.md [hunk 8]
```diff
@@ -1,95 +1,113 @@
...
+
+The group of elements is automatically wrapped in a `<fieldset>` element.
 
 <Doc::ComponentApi as |C|>
-  <C.Property @name="<[F].Label>" @type="yielded component">
-    It is a container that yields its content inside the `<label>` element. The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Label`](/components/form/base-elements/) component. _Notice: the `for` attribute of the label is automatically generated, using the `controlId` value of the control._
+  <C.Property @name="<[G].Legend>" @type="yielded component">
+    It is an (optional) container that yields its content inside the `<legend>` element. The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Legend`](/components/form/base-elements/) component.
```
Comment (jorytindall) [inline website/docs/components/form/radio/partials/code/component-api.md:36]:
```suggestion
    An optional container that yields its content inside the `<legend>` element. The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Legend`](/components/form/base-elements/) component.
```

File: website/docs/components/form/radio/partials/code/component-api.md [hunk 9]
```diff
@@ -1,95 +1,113 @@
...
   </C.Property>
-  <C.Property @name="<[F].HelperText>" @type="yielded component">
-    It is a container that yields its content inside the "helper text" block. The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::HelperText`](/components/form/base-elements/) component. _Notice: the `id` attribute of the element is automatically generated, using the `controlId` value of the control._
+  <C.Property @name="<[G].HelperText>" @type="yielded component">
+    It is a container that yields its content inside the "helper text" block (at group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::HelperText`](/components/form/base-elements/) component.
```
Comment (jorytindall) [inline website/docs/components/form/radio/partials/code/component-api.md:39]:
```suggestion
    A container that yields its content inside the "helper text" block (at group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::HelperText`](/components/form/base-elements/) component.
```

File: website/docs/components/form/radio/partials/code/component-api.md [hunk 10]
```diff
@@ -1,95 +1,113 @@
...
+    <br/><br/>
+    The `id` attribute of the element is automatically generated.
   </C.Property>
-  <C.Property @name="<[F].Error>" @type="yielded component">
-    It is a container that yields its content inside the "error" block. The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Error`](/components/form/base-elements/) component. _Notice: the `id` attribute of the `Error` element is automatically generated._
+  <C.Property @name="<[G].Radio::Field>" @type="yielded component">
+    It is used to yield one or more fields inside the group. For details about its API check the `Radio::Field` component above.
```
Comment (jorytindall) [inline website/docs/components/form/radio/partials/code/component-api.md:44]:
```suggestion
    Used to yield one or more fields inside the group. For details about its API check the `Radio::Field` component above.
```

File: website/docs/components/form/radio/partials/code/component-api.md [hunk 11]
```diff
@@ -1,95 +1,113 @@
...
+  </C.Property>
+  <C.Property @name="<[G].Error>" @type="yielded component">
+    It is a container that yields its content inside the "error" block (at group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Error`](/components/form/base-elements/) component.
```
Comment (jorytindall) [inline website/docs/components/form/radio/partials/code/component-api.md:47]:
```suggestion
    A container that yields its content inside the "error" block (at group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Error`](/components/form/base-elements/) component.
```

File: website/docs/components/form/radio/partials/code/component-api.md [hunk 12]
```diff
@@ -1,95 +1,113 @@
...
-  <C.Property @name="<[G].Legend>" @type="yielded component">
-    It is an (optional) container that yields its content inside the `<legend>` element. The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Legend`](/components/form/base-elements/) component.
-  </C.Property>
-  <C.Property @name="<[G].HelperText>" @type="yielded component">
-    It is a container that yields its content inside the "helper text" block (at group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::HelperText`](/components/form/base-elements/) component. _Notice: the `id` attribute of the element is automatically generated._
+  <C.Property @name="<[F].Label>" @type="yielded component">
+    It is a container that yields its content inside the `<label>` element. The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Label`](/components/form/base-elements/) component.
```
Comment (jorytindall) [inline website/docs/components/form/radio/partials/code/component-api.md:84]:
```suggestion
    A container that yields its content inside the `<label>` element. The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Label`](/components/form/base-elements/) component.
```

File: website/docs/components/form/radio/partials/code/component-api.md [hunk 13]
```diff
@@ -1,95 +1,113 @@
...
+    <br/><br/>
+    The `for` attribute of the label is automatically generated, using the `controlId` value of the control.
   </C.Property>
-  <C.Property @name="<[G].Radio::Field>" @type="yielded component">
-    It is used to yield one or more fields inside the group. For details about its API check the `Radio::Field` component above.
+  <C.Property @name="<[F].HelperText>" @type="yielded component">
+    It is a container that yields its content inside the "helper text" block. The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::HelperText`](/components/form/base-elements/) component.
```
Comment (jorytindall) [inline website/docs/components/form/radio/partials/code/component-api.md:89]:
```suggestion
    A container that yields its content inside the "helper text" block. The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::HelperText`](/components/form/base-elements/) component.
```

File: website/docs/components/form/radio/partials/code/component-api.md [hunk 14]
```diff
@@ -1,95 +1,113 @@
...
+    <br/><br/>
+    The `id` attribute of the element is automatically generated, using the `controlId` value of the control.
   </C.Property>
-  <C.Property @name="<[G].Error>" @type="yielded component">
-    It is a container that yields its content inside the "error" block (at group level). The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Error`](/components/form/base-elements/) component. _Notice: the `id` attribute of the `Error` element is automatically generated._
+  <C.Property @name="<[F].Error>" @type="yielded component">
+    It is a container that yields its content inside the "error" block. The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Error`](/components/form/base-elements/) component.
```
Comment (jorytindall) [inline website/docs/components/form/radio/partials/code/component-api.md:94]:
```suggestion
    A container that yields its content inside the "error" block. The content can be a simple string, or a more complex/structured one (in which case it inherits the text style). For details about its API check the [`Form::Error`](/components/form/base-elements/) component.
```

File: website/docs/components/form/radio/partials/code/how-to-use.md [hunk 15]
```diff
@@ -83,11 +84,14 @@ You can add extra information to the group using an "helper" text:
 </Hds::Form::Radio::Group>
 ```
 
-When the "helper" text is added, the component automatically adds an `aria-describedby` attribute to the `fieldset`, associating it with the automatically generated `ID` of the helper text element.
+#### Extra content in legend and helper text
 
-##### Extra content in legend and helper text
+The `Legend` and `HelperText` contextual components used in the "group" are yielding their content: this means you can pass not just plain text, but also structured content.
```
Comment (jorytindall) [inline website/docs/components/form/radio/partials/code/how-to-use.md:89]:
```suggestion
The `Legend` and `HelperText` contextual components used in the "group" yield their content: meaning you can pass not just plain text, but also structured content.
```

File: website/docs/components/form/radio/partials/code/how-to-use.md [hunk 16]
```diff
@@ -129,13 +131,9 @@ It's possible to add a visual indication if a group is "required" or is "optiona
...
 
-_Notice: for complex forms we suggest to indicate **required** fields, since this is the most explicit and transparent method and ensures users don’t have to make assumptions. For shorter, simpler forms (ie. login/signup and feedback requests) we suggest to indicate **optional** fields._
-
-##### Validation
-
-Note: the validation of the form fields is entirely delegated to the "consumer" of the HDS components. What we provide is the visual representation of an invalid state of the field at UI level. When and how to provide this visual feedback to the user is responsibility left to the developer.
+#### Validation
 
-To show the user that their input is not valid, you have to provide an error message (using the `Error` contextual component):
+To show the user that their input is not valid, you have to provide an error message (using the `Error` contextual component).
```
Comment (jorytindall) [inline website/docs/components/form/radio/partials/code/how-to-use.md:136]:
```suggestion
If an input is not valid, provide the user with an error message using the `Error` contextual component.
```

**Review by jorytindall — APPROVED** [review]
Let a couple of grammatical suggestions

**Comment by alex-ju** [comment]
> Let a couple of grammatical suggestions

Excellent points there, Jory! I tried applying these updates across the repo except on the components that have PRs on the fly (select, textarea, toggle) to avoid creating conflicts.

**Pull request merged**